cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/ResultsBinding.java

```diff
@@ -76,11 +76,18 @@ public String resolveResult(Map<String, Serializable> inputs,
         // In the case of operation, we resolve the result by searching for the first result with a true expression
         // An empty expression passes as true
         for(Result result : possibleResults){
-            String expression = result.getExpression();
+            Serializable rawExpression = result.getExpression();
+
             // If the answer has no expression, we treat it as a true expression, and choose it
-            if(StringUtils.isEmpty(expression)) {
+            if(StringUtils.isEmpty(rawExpression)) {
                 return result.getName();
             }
+
+            String expression = extractExpression(rawExpression);
+            if (expression == null) {
```

Comment:
From the docs: 

The first result in the list whose expression evaluates to true, or does not have an expression at all, will be passed back to the calling task to be used for navigation purposes.